### PR TITLE
[Bugfix] Enable padded FP4 quantization

### PR DESCRIPTION
**Pull request by roikoren755**
## Purpose
This PR fixes an issue with padded FP4 quantization, where previously initialized values in the allocated tensors aren't overwritten properly when quantizing to FP a tensor which requires padding.

This issue prevents running NVIDIA's new Nemotron-Nano-9B-v2 with TP2, as some of the tensor sizes in such cases do not divide evenly by 128.

Note that this specific model doesn't work with TP>2, and this PR doesn't solve issues that come up in those scenarios. For TP4 the issue comes from inside the FP4 GEMM kernel, where it expects some matrix size to be divisible by 32, and for TP8 the issue happens when creating the model weights, where some layers' in features are not divisible by 16. The same issues happen in TensorRT-LLM as well.

## Test Plan
No need for new tests, as tests are already in place to verify FP4 quantization, both with and without padding.

## Test Result
All existing FP4 quantization tests passed locally.

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/_custom_ops.py [hunk 1]
```diff
@@ -1374,7 +1374,7 @@ def scaled_fp4_quant(
     )
 
     # Two fp4 values will be packed into an uint8.
-    output = torch.empty((m, n // 2), device=device, dtype=torch.uint8)
+    output = torch.zeros((m, n // 2), device=device, dtype=torch.uint8)
```
Comment (pavanimajety) [inline vllm/_custom_ops.py]: When a tensor of required out shape is allocated, why do we also need to initialize it with zeros? The divisibility check is only on the reducing dimension, could you clarify this?

**Comment by pavanimajety** [comment]
Thanks for the PR! Could you also post lm_eval results for any other FP4 model to ensure that previous paths don't break?

File: vllm/_custom_ops.py (hunk 1 above)
Comment (roikoren755) [reply vllm/_custom_ops.py]: Did it for consistency's sake, but you are correct, this allocation can stay as `torch.empty`. I'll revert this line.

**Review by pavanimajety — APPROVED** [review]

**Review by mgoin — APPROVED** [review]
